src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -119,15 +119,15 @@ GTCodeGen::IntervalDefinitions::IntervalDefinitions(const iir::Stencil& stencil)
 /// every field is a template argument.
 class StencilFunctionAsBCGenerator : public ASTCodeGenCXX {
 private:
-  std::shared_ptr<sir::StencilFunction> function;
-  const std::shared_ptr<iir::StencilInstantiation> instantiation_;
+   const std::shared_ptr<iir::StencilFunctionInstantiation>& function_;
+  //  const std::shared_ptr<iir::StencilInstantiation> instantiation_;
+  const iir::IIR* iir_;
```

Comment:
I think const std::shared_ptr<>& would be better than iir* ? better not getting out of smart ptr into raw ptr